### PR TITLE
Add Call/Room/Click events and tidy up for Swift

### PR DESCRIPTION
**Pull request by pixlwave**
Adds call events as detailed in the Analytics Events spreadsheet as well as making various tweaks for Swift:
- Add titles to `eventName` enums. Without these I see compile errors due to conflicting names.
- Tweak the quicktype parameters in the swift build script to make more usable.
- Add a basic podspec to import the generated files into the project (this will probably need more work before it is ready).

On the `CallError` event: Is if the idea is to send this alongside a regular `Error` event with details about what went wrong?

**Update:** Adds Click, CreatedRoom and JoinedRoom events too. The click event has a placeholder UI element for now, to allow the generator to work (not intended to be permanent).

**Update 2:** Removed the enum titles, will continue with Swift refinements in a separate branch.

### CODE REVIEW

File: schemas/Error.json [hunk 1]
```diff
@@ -3,17 +3,20 @@
...
       "type": "string",
+      "title": "ErrorDomain",
       "enum": [
         "E2EE",
         "VOIP"
       ]
     },
     "name": {
       "type": "string",
+      "title": "ErrorName",
```
Comment (pixlwave) [inline schemas/Error.json:19]: This and and the one on the domain were to reduce potential name clashes in the future.

File: schemas/Error.json [hunk 2]
```diff
@@ -3,17 +3,20 @@
   "description": "Triggered when an error occurred",
   "properties": {
     "eventName": {
+      "title": "EventNameError",
```
Comment (bmarty) [inline schemas/Error.json:6]: Why not using "Error" as a prefix like for `ErrorDomain` and `ErrorName`?

**Review by bmarty — COMMENTED** [review]
One small remark, maybe applicable to the other types

File: schemas/Error.json (hunk 2 above)
Comment (pixlwave) [reply schemas/Error.json]: Updated fwiw.

File: schemas/CallEnded.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "placed": {
+      "description": "Whether this user placed it.",
+      "type": "boolean"
+    },
+    "isVideo": {
+      "description": "Whether its a video call or not.",
+      "type": "boolean"
+    },
+    "durationMs": {
+      "description": "The duration of the call.",
```
Comment (novocaine) [inline schemas/CallEnded.json:18]: description should probably say its in ms

**Comment by novocaine** [comment]
It's unfortunate the the enums get generated without a namespace or inside the class - this is how the TS generator works. Would it be legal to do that, but quicktype doesn't?

**Comment by pixlwave** [comment]
> It's unfortunate the the enums get generated without a namespace or inside the class - this is how the TS generator works. Would it be legal to do that, but quicktype doesn't?

Yep, nested enums would would be the preferable way to do this. I'm going to investigate going down the same route as Benoit to generate the types ourselves, as this will help a lot. For now I'll remove the namespacing workaround from the PR so it can be merged and then tackle the Swift stuff separately.

**Review by novocaine — APPROVED** [review]

**Comment by novocaine** [comment]
:shipit:

**Pull request merged**